semgrep-core/src/reporting/JSON_report.ml

```diff
@@ -101,6 +101,42 @@ let range_of_any_opt startp_of_match_range any =
       let startp, endp = position_range min_loc max_loc in
       Some (startp, endp)
 
+let metavar_string_of_any any =
```

Comment:
Maybe add a `TODO` to handle the cases where there is no origin tokens? E.g. in `x = 1; y = x + 1; ...` we have `y = 2` but there is no source location for `2`.